### PR TITLE
Application logo on splash screen

### LINKED ISSUE #230: Use the logo + integrate splash screen
Acceptance criteria:
- The app logo must be the logo on the shared drive
- The picture on the login page must be the logo on the shared drive
- The PenPot must be updated accordingly
- An app splash screen must be added with the new logo

### PR DESCRIPTION
**Pull request by gruvw**
Fixes #230 

### Application logo and splash screen

This PR makes use of the new application logo.

It also adds a splash screen to the app in order to improve the initial loading screen of the app (which was plain white before).

#### Acceptance criteria

- [X] The correct logo must be used on the PenPot
- [X] Use the blue from the logo as our theme seed color
- [X] The app logo must be the logo on the shared drive (white version)
- [X] The picture on the login page must be the logo on the shared drive
- [X] An app splash screen must be added with the new logo
- [x] App name should be "Proxima"

#### Result

Application logo on android home screen:

**Note**: the app name is now "Proxima" (not "proxima") since 13b81b5f7507cf6f607dbcab3e2566522565c06c.

Application logo on login screen:

Splash Screen:

#### Pubspec modifications

- Added the `flutter_launcher_icons` https://pub.dev/packages/flutter_launcher_icons package to automatically generate the android app icons based on the asset file
- The above package is configured through the `flutter_launcher_icons` section on the `pubspec.yaml` file
- Added the `flutter_native_splash` https://pub.dev/packages/flutter_native_splash package to automatically generate a native splash screen base on the application icon
- The above package is configured through the `flutter_native_splash` section on the `pubspec.yaml` file

**Note**: the introduction of the splash screen handling must be done in `main.dart` so it cannot be properly tested. Therefore, this PR costs -0.1% coverage :(

### CODE REVIEW

**Comment by gruvw** [comment]
#### Information / Question

Currently, the splash screen is well displayed and can be checked on debug mode in the emulator.

In release mode and using a modern phone, the splash screen goes away very fast and we see a small white flash.
I am wondering if we should intentionally add a small loading time (like 0.6 seconds) to have a splash screen displayed even on fast phones.

File: pubspec.yaml [hunk 1]
```diff
@@ -40,9 +40,19 @@ dev_dependencies:
...
     - assets/icons/
+
+# Command to generate the icons: `dart run flutter_launcher_icons`
+flutter_launcher_icons:
+  android: true
+  image_path: assets/images/proxima_white.jpg
+
+# Command to generate the splash screen: `dart run flutter_native_splash:create`
+flutter_native_splash:
+  color: "#FFFEFE"  # logo background color
```
Comment (CHOOSEIT) [inline pubspec.yaml:58]: Shouldn't also set the `color` for the `android12:` ?

**Review by CHOOSEIT — COMMENTED** [review]
Great PR, great work 💯 .

Thank you !

This is not in your acceptances criteria, but we also have the logo in dark mode, shouldn't we adapt the logo depending on the theme? Similarly I saw that on the `flutter_native_splash` package we could specify that the light and dark mode logo and color, shouldn't we do that ?

**Comment by CHOOSEIT** [comment]
> I am wondering if we should intentionally add a small loading time (like 0.6 seconds) to have a splash screen displayed even on fast phones.

I also think that it could be a good idea, I think that we could just wait that the user automatically loggings in. (Currently the user see the home page. So the idea could be to wait until the `isUserLoggedInProvider` is completed for the first time ?

(Noting that by preserving the splash screen and waiting for the firebase base to initialise as you did seems to reduce the waiting time before being navigated to the home screen 👍🏼 )

File: pubspec.yaml (hunk 1 above)
Comment (gruvw) [reply pubspec.yaml:58]: No, this is if we need a special one. Currently, it works on android12 too.

File: lib/views/pages/login/login_page.dart [hunk 2]
```diff
@@ -11,7 +11,7 @@ class LoginPage extends ConsumerWidget {
 
   static const tagLineText = "Discover the world,\n one post at a time.";
 
-  static const _logoAsset = "assets/images/proxima_logo.jpeg";
+  static const _logoAsset = "assets/images/proxima_white.jpg";
```
[the remark this answers is not in the document]
Comment (CHOOSEIT) [reply lib/views/pages/login/login_page.dart:14]: Ahah ! I missed the first line 😢 I was also looking at the wrong mockup; everything is fine. Thank you! 👍🏼

**Comment by gruvw** [comment]
Ok, so I applied your suggestions + added the splash delay.

I also converted the logo image to use `png` instead of `jpg` because of a weird generation issue... Now it works well on my phone and on the emulator :+1:

**Comment by CHOOSEIT** [comment]
> This is not in your acceptances criteria, but we also have the logo in dark mode, shouldn't we adapt the logo depending on the theme? Similarly I saw that on the `flutter_native_splash` package we could specify that the light and dark mode logo and color, shouldn't we do that ? (I saw it was specified white mode so seems ok then and maybe to add it later ?)

We decided to keep the light logo for the moment 😄 💪🏼 

Great work thank you !

**Review by CHOOSEIT — APPROVED** [review]
LGTM ! 💯 

Great work !

Thank you 💪🏼 🎉

**Pull request merged**